src/aria/utils/Array.js

```diff
@@ -173,6 +173,31 @@ var Aria = require("../Aria");
                     }
                 }
                 return output;
+            },
+            /**
+             * This tests whether all elements in the array pass the test given by the predicate callback function. 
+             * As long as the predicate (callback) returns truthy values, the loop will continue. However, as soon as a falsy
+             * value is returned, the function will return without analyzing the rest of the elements.
+             * @param {Array} array Array to analyze.
+             * @param {Function} Function to test each element of the array. This function receives as arguments the
+             * value, the index and the array.
+             * @param {Object} thisObject Object to use as this when executing callback.
+             * @return {Boolean} true if every element in the given array returns truthy values for the predicate (callback).
+             */
+            every : (!arrayPrototype.every) ? function(array, callback, thisObject) {
+                // clone array to avoid mutation when executing the callback
```

Comment:
This comment — which I know you copied, no blame on you — actually got me confused. I looked at the code too fast then, and thought that this array was passed as the third parameter to the `callback`. Hence my comments for the documentation and for the setting of the default value for `thisObject` which are actually wrong.

I would suggest being a bit more precise and say something more like `to prevent being impacted by possible mutations of the array in the callback`.